pages/settings/wallets/ConnectWallet.tsx

```diff
@@ -0,0 +1,281 @@
+import Loading from "~/components/Loading";
+import Screen from "~/components/Screen";
+import { Button } from "~/components/ui/button";
+import { Text } from "~/components/ui/text";
+import { useGetFiatAmount } from "~/hooks/useGetFiatAmount";
+import { DEFAULT_WALLET_NAME } from "~/lib/constants";
+import { errorToast } from "~/lib/errorToast";
+import { useAppStore } from "~/lib/state/appStore";
+
+export function ConnectWallet() {
+  const { appicon, appname, callback } = useLocalSearchParams<{
+    appicon: string;
+    appname: string;
+    callback: string;
+  }>();
+  const getFiatAmount = useGetFiatAmount();
+  const [isLoading, setLoading] = React.useState(false);
+  const wallets = useAppStore((store) => store.wallets);
+  const selectedWalletId = useAppStore((store) => store.selectedWalletId);
+
+  const [nwcUrl, setNwcUrl] = React.useState<string | null>(null);
+  const [redirectCountdown, setRedirectCountdown] = React.useState<
+    number | null
+  >(null);
+
+  const leftPlugAnim = React.useRef(new Animated.Value(0)).current;
+  const rightPlugAnim = React.useRef(new Animated.Value(0)).current;
+
+  React.useEffect(() => {
+    if (nwcUrl) {
+      Animated.parallel([
+        Animated.timing(leftPlugAnim, {
+          toValue: 5,
+          duration: 500,
+          useNativeDriver: true,
+          easing: Easing.ease,
+        }),
+        Animated.timing(rightPlugAnim, {
+          toValue: -15,
+          duration: 500,
+          useNativeDriver: true,
+          easing: Easing.ease,
+        }),
+      ]).start();
+    } else {
+      leftPlugAnim.setValue(0);
+      rightPlugAnim.setValue(0);
+    }
+  }, [nwcUrl, leftPlugAnim, rightPlugAnim]);
+
+  React.useEffect(() => {
+    if (redirectCountdown === null || !nwcUrl) {
+      return;
+    }
+
+    if (redirectCountdown === 0) {
+      (async () => {
+        const callbackUrl = `${callback}?value=${nwcUrl}`;
+        try {
+          await openURL(callbackUrl);
+        } catch (error) {
+          console.error(error);
+          errorToast(
+            new Error("Couldn't open URL, do you have the app installed?"),
+          );
+        }
+        if (router.canDismiss()) {
+          router.dismissAll();
+        }
+        router.replace("/");
+      })();
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setRedirectCountdown((prev) => (prev ? prev - 1 : prev));
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [redirectCountdown, nwcUrl, callback]);
+
+  const confirm = async () => {
+    setLoading(true);
+    try {
+      const nwcClient = useAppStore.getState().nwcClient;
+      if (!nwcClient) {
+        throw new Error("NWC client not connected");
+      }
+      const secretKey = generateSecretKey();
+      const pubkey = getPublicKey(secretKey);
+      {
+        /* TODO: REPLACE WITH BUDGET INFO (AND METHODS?) */
+      }
+      const response = await nwcClient.createConnection({
```

Comment:
can update to use `@getalby/sdk 4.1.0-beta.0` and update the params here to be consistent with the http deeplink flow